judge/judgedaemon.main.php

```diff
@@ -529,6 +534,13 @@ function fetch_executable_internal(
     $runuser .= '-' . $options['daemonid'];
 }
 
+if ($runuser === posix_getpwuid(posix_geteuid())['name'] ||
+    RUNGROUP === posix_getgrgid(posix_getegid())['name']
+) {
+    echo "Do not run the judgedaemon as the runser or rungroup.\n";
```

Comment:
Can we instead of what not to do say what to do, i.e. do we know at this point as which user this should be run with?

Reply:
I'm not sure we know that, basically it needs to be any user except the specials 'runuser' and 'root'.